src/pylyrics/extract_lyrics.py

```diff
@@ -0,0 +1,219 @@
+# author: Macy Chan
+# date: 2022-01-14
+
+import requests
+import os, sys
+import pandas as pd
+import json
+import urllib.parse
+import re
+import lyricsgenius
+from alive_progress import alive_bar
+
+
+class HiddenPrints:
+    def __enter__(self):
+        """
+        Suppress printing output
+        """
+        self._original_stdout = sys.stdout
+        sys.stdout = open(os.devnull, "w")
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """
+        Exit suppress printing output
+        """
+        sys.stdout.close()
+        sys.stdout = self._original_stdout
+
+
+def get_lyrics_from_Genius(genius, title, artist):
+    """
+    Find lyrics of song using Genius API
+
+    Parameters
+    ----------
+    title : string
+        song title
+    artist : string
+        artist of song
+
+    Returns
+    ----------
+    lyrics : string
+        lyrics of song
+    """
+    with HiddenPrints():
+        song = genius.search_song(title, artist)
+        if song:
+            lyrics = song.lyrics
+            return lyrics
+
+
+def extract_lyrics(token, songs):
+    """
+    Extracting lyrics for a maximum of 5 songs
+
+    Parameters
+    ----------
+    songs : dataframe
+        A dataframe consists two columns: song_title and artist
+    token : string
+        A token from Genius for asccessing lyrics
+
+    Returns
+    ----------
+    songs : dataframe
+        Return lyrics of song with three columns: song_title, artist and lyrics
+    """
+    try:
+        genius = lyricsgenius.Genius(token, retries=5)
+
+        if songs.empty:
+            raise ValueError("Empty DataFrame")
+
+        if len(songs) > 5:
+            raise ValueError("Only accept song lists short than 5 songs or below")
+
+        if not ("song_title" in songs and "artist" in songs):
+            raise ValueError(
+                "Invalid column names, please provide song_title and artist"
+            )
+
+        if not (
+            songs["song_title"].dtypes == "object"
+            and songs["artist"].dtypes == "object"
+        ):
+            raise ValueError(
+                "Invalid column names, please provide song_title and artist"
+            )
+
+        songs["lyrics"] = ""
+
+        print("Checking URL connection...")
+        with alive_bar(len(songs), bar="bubbles", spinner="notes2") as bar:
+            for i in range(len(songs)):
+                lyrics = get_lyrics_from_Genius(
+                    genius,
+                    songs.iloc[i]["song_title"],
+                    songs.iloc[i]["artist"],
+                )
+
+                if lyrics:
+                    songs["lyrics"][i] = lyrics
+                bar()
+
+        return songs
+
+    except Exception as req:
+        print(req)
+
+
+""" --For later milestones
+# test
+# /data/credentials.json
+# {
+#   "token": "YOUR-TOKEN"
+# }
+
+with open("data/credentials.json") as f:
```

Comment:
Shouldn't we include the token reading to `extract_lyrics()`?

Reply:
Token should be provided by user. It is private information.

Reply:
Agreed. What I am saying is that the user provides the location where credentials.json is located and we then read it into a token variable to be used for reading lyrics.

Reply:
the user can just give use the token as string, it is up to them on how to store their token